src/providers/token-provider.ts

```diff
@@ -627,6 +628,64 @@ export class TokenProvider implements ITokenProvider {
   ) {
   }
 
+  private async getTokenSymbol(addresses: string[], providerConfig?: ProviderConfig) {
+    let result;
+
+    try {
+      result = await this.multicall2Provider.callSameFunctionOnMultipleContracts<
+        undefined,
+        [string]
+      >({
+        addresses,
+        contractInterface: IERC20Metadata__factory.createInterface(),
+        functionName: 'symbol',
+        providerConfig,
+      });
+    } catch (error) {
+      log.error(`TokenProvider.getTokenSymbol failed with error ${error}. Trying with bytes32.`);
+
+      const bytes32Interface = new Interface([
+        {
+          inputs: [],
+          name: 'symbol',
+          outputs: [
+            {
+              internalType: 'bytes32',
+              name: '',
+              type: 'bytes32',
+            },
+          ],
+          stateMutability: 'view',
+          type: 'function',
+        }
+      ]);
+
+      result = this.multicall2Provider.callSameFunctionOnMultipleContracts<
```

Comment:
Hmm.. thinking through this more, does this mean we retry the whole batch with bytes32? Then wouldn't the other tokens start failing?

Reply:
Yea this would cause other standard tokens to fail

Reply:
oh boy, that's a good call, there's really no way of knowing if there's one bad token in there that spoils the whole bunch, right?

Reply:
I mean, either way it's gonna fail right? but this would still work for when we are fetching only 1 token and that token is wrong

Reply:
I ran some queries, and the majority of times this runs for only 1 address, so it should work for those cases